### ISSUE TITLE
Setting a big project default font makes GUT unusable

### ISSUE THREAD

**Issue by nightblade9** [bitwes/Gut#283]
If I set a large default font on my project theme (for accessibility reasons), the GUT UI becomes huge and it's difficult/impossible to use (can't select a particular test suite). If I apply a new theme to the Gut control, it looks OK in the editor, but doesn't apply in-game. (This is Godot 3.2.3.)

![image](https://user-images.githubusercontent.com/31222662/110565881-33eae980-811d-11eb-97c0-07e07429beb4.png)

![image](https://user-images.githubusercontent.com/31222662/110565936-482ee680-811d-11eb-95e3-9c9976bb3e81.png)

**Comment by Technohacker** [comment]
Without applying any themes on the GUT tree, the GUI is visibly distorted (v7.1.0). I tried looking through the remote scene tree and noticed quite a few Controls had rect_scale set to (2, 2). Setting them back to (1, 1) seems to fix them. I'm still going through all of them

**Comment by Technohacker** [comment]
@bitwes sorry for the ping but, is there any particular reason for not using BoxContainers in the GUI? If not, I am in the process of restructuring the GUI on my local copy of the addon to use containers. It could alleviate the need to handle custom resize and window movement code, and you wouldn't need to have a placeholder for the editor. If time permits and you're willing, maybe I could make a PR for the same?

**Comment by bitwes** [comment]
No reason at all.  That GUI is SUPER old.  I didn't know about box
containers when I started making it (possible they didn't exist, idk).
There's a lot of room for improvement there.  A PR would be great.

**Comment by Technohacker** [comment]
Interesting, for me the GUI having a higher scale than expected happens to be due to HiDPI being enabled. I'll leave this here for the record in case someone else is facing the same issue

**Comment by nightblade9** [comment]
Any chance we can resolve this PR quickly by simply specifying a font on the root node of the Gut UI element? I use large fonts for accessibility reasons and this makes it painful to use.

**Comment by bitwes** [comment]
That should work.  `GutScene.gd` is what needs to change.  The problem is that the gui shown in the editor is just a placeholder.  It punches through some of the properties at runtime but not all.  There is probably a way to just export all the settings set and apply them to the runtime instance.  `plugin_control.gd` creates an instance for the editor and `gut.gd` creates its own instance when tests are run.

You can manually make changes to `GutScene.tscn`.  These will be overwritten on update, but the changes should be minimal until a fix is in place.  

The gui is relic of the early days when GUT was a single script (before plugins or before I knew about them the whole gui was made with code).  It's been improved to at least be a scene now, but it leaves a lot to be desired.

**Comment by nightblade9** [comment]
Looks like it doesn't work; the root node doesn't have a font property. I manually set the font for all the controls inside the GUT scene, that works fine for now. You can probably do something similar when you fix this.

**Comment by Technohacker** [comment]
@nightblade9 could you test out my PR (#292) for this issue to see if it works?

**Comment by nightblade9** [comment]
I'm not sure. Am I doing something wrong? This is what the GUI looks like to me:

![image](https://user-images.githubusercontent.com/31222662/132712857-85dc58a5-bf2f-4ed0-aa7d-b6f0e798b97d.png)

What I did:

- `git clone git@github.com:technohacker/gut`
- `cd gut`, `git checkout gui_refactor`
- Run `git log` and verify your commit is present
- Copy the `addons/gut` folder from your repo, over the `addons/gut` folder in my game repo
- Verify that my game diff shows `GutScene.tscn` changed
- Delete and recreate the scene by picking new node > `Gut`
- Re-adding all my settings
- Set the font size to 8px (just to test)

The scene looks fine in-editor, but squashes up into the corner, and still appears to use the default font - not the 8px Anonymous Pro that the settings list.

**Comment by Technohacker** [comment]
Ah ouch, what if you set the Gut node to full rect size?

**Comment by nightblade9** [comment]
Forgot to mention - I set the margin to match full-screen size, and also set the minimum size to match. Those fixed it in the editor view, but not at runtime.

**Comment by nightblade9** [comment]
Any updates on this?

**Comment by bitwes** [comment]
I've got this on the 7.30 milestone (next release).  I have not looked into further.  

7.2.0 added the GUT Panel to run tests w/o a scene which might help as a temporary workaround.  Since you can see the output in the GUT Panel, you could set the opacity to hide GUT when running tests so you at least don't have to look at it.

**Comment by nightblade9** [comment]
While it sucks, I finally found a work-around that makes the list of test-cases readable.

- Create a new scene with a `Control` as the root node
- Add a `GUT` node as a child of root
- Add a new theme to the root node, with a default font that's a small size

The list of test-cases is finally readable, at least. Note that having the GUT node as the root node and applying the theme directly to it, doesn't help; nor does changing the `font_size` property - the dropdown listing test-cases doesn't seem to use that value.

Also note: having a small window size (640x360 in my case) truncates the GUT window. I probably won't make a game with this small a size + this large a font in the future (it's painful in many ways).

**Comment by bitwes** [comment]
@nightblade9 I've merged a fix for this into master.  Can you verify the fix?

**Comment by nightblade9** [comment]
TL;DR I don't think this fixes the problem, but maybe I did something wrong. Here's my screenshot:

![image](https://user-images.githubusercontent.com/31222662/147517124-4dd6ae9c-5e99-407d-bfc5-7c4554384333.png)

Repro steps:

- Download Gut source code from Git
- Unzip to my project root directory and confirm that I'm overwriting lots of files
- Check Git's diff and see lots of files changed, e.g. `gut.gd`
- Delete and recreate a node that's "GUT"
- Set the font to Default and the font-size to 8
- Run

The text still looks really big. As you can see, with my crazy pathing, you can't tell which test case is which.

Here's how it looks with my work-around applied. (Note that my work-around doesn't work any more with the latest git version of Gut.)

![image](https://user-images.githubusercontent.com/31222662/147517223-a793e12d-767f-4b51-9ed1-946750792722.png)

**Comment by bitwes** [comment]
Thanks for testing it out.  I think we can confirm this fixes the issue, since the larger font is not being applied.  This is also why your workaround doesn't work.  The scene now has its own theme and font which prevents any higher level font from being applied.

I used the default font size of 16, and just changed it to 14 (anything smaller than 14 was hard to read on my setup).  I may change the font more before release, but at least the issue is fixed.  Thanks again.

**Comment by nightblade9** [comment]
Makes sense. It would be great if the font size was customizable (16 works for you, 14 works for me). I guess that can be another issue / future change.
